### ISSUE TITLE
Add shell kconfig API

### ISSUE THREAD

**Issue by xuyang0410** [linux-test-project/ltp#891]
A month ago, I  recommended xfstests refer  to ltp C kconfig API [1] and this patch was merged into xfstest community.  I think we should introduce shell kconfig parser api into tst_test.sh  and we can refer to xfstests kconfig api[2]. Any idea? 

[1][Re: [PATCH v5] ext4: add test for all ext4/ext3/ext2 mount options](https://www.spinics.net/lists/fstests/msg17862.html)
[2][common/rc: add _require_kernel_config and _has_kernel_config](https://git.kernel.org/pub/scm/fs/xfs/xfstests-dev.git/commit/?id=d83eaf7464790217c12a0f4e3ed9064f6bbc2540)

**Comment by metan-ucw** [comment]
If we are going to add it can we please do it as:

- The kconfig strings would be stored in numbered TST_REQUIRE_KCONFIG variables
- The test library will pass these to a tst_kconfig_check binary would call the tst_kconfig_check

the shell library code would look like:

```
tst_kconfig_check "TST_REQUIRE_KCONFIG1" "TST_REQUIRE_KCONFIG2" ....
if [ $? -ne 0]
        exit $?
```

I guess that passing them up to `TST_REQUIRE_KCONFIG10` would be enough for everyone. We can also explicitly warn if TST_REQUIRE_KCONFIG11 is defined so that if someone needs more than 10 it will tell him to adjust the library code.

And the tst_kconfig_check would just pass the argv+1 to the tst_kconfig_check(), should be as easy as:

```
int main(int argc, char *argv[])
{
        tst_kconfig_check(argv+1);

        exit(0);
}

**Comment by pevik** [comment]
I suppose we cannot have it all in single `TST_REQUIRE_KCONFIG` variable and separate by space because single kconfig pattern can contain spaces, right? (e.g. `CONFIG_MMU & CONFIG_EXT4_FS=m`).

How about have a default separator (e.g. space) and allow to overwrite it with another variable (e.g. TST_REQUIRE_KCONFIG_IFS`). I mean similar way as `TST_TEST_DATA` and `TST_TEST_DATA_IFS` in tst_test.sh.

**Comment by metan-ucw** [comment]
@pevik Exactly we would have to choose a different separator than whitespace since expressions can contain these. Also if we decide to use separator we can just set use `#` since that will not appear in the expressions.

**Comment by pevik** [comment]
I'd be for `'#` then.
Also, this issue has been reported some time ago https://github.com/linux-test-project/ltp/issues/700, can we close one of them?

**Comment by xuyang0410** [comment]
Use "#" looks strange because  kernel config file uses # to undefine those kconfig.  How about using `,`, something like `CONFIG_MMU  |   CONFIG_EXT4_FS=m ,  CONFIG_QUOTACTL=y`

**Comment by xuyang0410** [comment]
I have done this but miss some cases and documentation. So I will send this on next week.

**Comment by xuyang0410** [comment]
It has been Implemented in [e3e5c71abfc](https://github.com/linux-test-project/ltp/commit/e3e5c71abfc0b59dbfe5874836346516b4f0746a) and [cc135d9c](https://github.com/linux-test-project/ltp/commit/cc135d9c8504b6ce1dcc78d5cdbc87bd67cdecc7)
